### PR TITLE
model(cdc): add a function to recover TiDB TableInfo

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/3242, a part of https://github.com/pingcap/tiflow/pull/4587

### What is changed and how it works?

after recover TiDB TableInfo from CDC's redo log or RowChangedEvent, we can use sqlmodel and generate DML in the same way as DM

#### Tests 

 - Unit test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/entry/mounter_test.go [hunk 1]
```diff
@@ -1087,3 +1094,72 @@ func TestDecodeEventIgnoreRow(t *testing.T) {
...
+				") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin",
+		},
+		{
+			"CREATE TABLE t1 (" +
+				" c INT UNSIGNED," +
+				" c2 VARCHAR(10) NOT NULL," +
+				" c3 BIT(10) NOT NULL," +
+				" UNIQUE KEY (c2, c3)" +
+				")",
+			// CDC discards field length, charset and collation.
```
Comment (lance6716) [inline cdc/entry/mounter_test.go:1117]: maybe we can fill utf8mb4 since TiKV uses this stored format?

**Comment by lance6716** [comment]
ptal @asddongmen

**Review by asddongmen — APPROVED** [review]

File: cdc/model/sink.go [hunk 2]
```diff
@@ -442,6 +443,97 @@ type RedoColumn struct {
...
+		for _, offset := range colOffsets {
+			col := ret.Columns[offset]
+
+			indexCol := &model.IndexColumn{}
+			indexCol.Name = col.Name
+			indexCol.Offset = offset
+			indexInfo.Columns = append(indexInfo.Columns, indexCol)
+		}
+
+		// TODO: revert the "all column set index related flag" to "only the
```
Comment (zhaoxinyu) [inline cdc/model/sink.go:529]: Does this TODO item need to be done in this PR?

Comment (lance6716) [reply cdc/model/sink.go:529]: No, I'll test if this reversion is really needed in future PR

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**